> What warning are you seeing on which compiler? I don't think we see that internally. Also, C-style casts are discouraged.

`"INFO: From Compiling tensorflow/core/lib/strings/proto_serialization.cc [for host]:
tensorflow/core/lib/strings/proto_serialization.cc: In function 'bool tensorflow::SerializeToBufferDeterministic(const google::protobuf::MessageLite&, char*, size_t)':
tensorflow/core/lib/strings/proto_serialization.cc:75:44: warning: comparison of integer expressions of different signedness: 'size_t' {aka 'long unsigned int'} and 'int' [-Wsign-compare]
   75 |   return !output_stream.HadError() && size == output_stream.ByteCount();
      |                                       ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
"`

Can you recommend an appropriate change?

--- a/tensorflow/core/lib/strings/proto_serialization.cc
+++ b/tensorflow/core/lib/strings/proto_serialization.cc
@@ -72,7 +72,7 @@ bool SerializeToBufferDeterministic(const protobuf::MessageLite& msg,
   protobuf::io::CodedOutputStream output_stream(&array_stream);
   output_stream.SetSerializationDeterministic(true);
   msg.SerializeWithCachedSizes(&output_stream);
-  return !output_stream.HadError() && size == (size_t)(output_stream.ByteCount());
+  return !output_stream.HadError() && size == static_cast<size_t>(output_stream.ByteCount());
 }
 
 bool AreSerializedProtosEqual(const protobuf::MessageLite& x,

